### PR TITLE
Adds email send

### PR DESCRIPTION
**Pull request by chrisfoster76**

### CODE REVIEW

File: src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Services/ProviderEmailNotificationServiceTests/WhenSendingProviderCommitmentStopNotification.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            var payload = new Apprenticeship
+            {
+                ProviderId = 123L,
+                LegalEntityName = "Legal Entity Name",
+                ULN = "1234567890",
+                FirstName = "John",
+                LastName = "Smith",
+                StopDate = new DateTime(2018, 05, 01)
+            };
+            _exampleApprenticeship =
```
Comment (lazcool) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application.UnitTests/Services/ProviderEmailNotificationServiceTests/WhenSendingProviderCommitmentStopNotification.cs]: Why the serialization round-trip?
If we want to test if the serialization round-trip is successful, it should be in its own test.

**Comment by lazcool** [comment]
I've checked the templates against the tokens supplied in the code, and I noticed case differences.
I'm guessing the tokens are matched case-insensitive, but I can't find any mention of token case-insensitivity in the Gov.Uk Notify online documentation, so it might be prudent to case match.

Also, (and this is probably out-of-scope for this story,) even though the tokens only need to match between the template and the code referencing that template, it might be worthwhile being consistent with token names between templates, to facilitate possible code-reuse when generating tokens, especially now we have centralised email services.

File: src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/UpdateApprenticeshipStatus/UpdateApprenticeshipStatusCommandHandler.cs [hunk 2]
```diff
@@ -52,6 +60,9 @@ public UpdateApprenticeshipStatusCommandHandler(IEmployerCommitmentApi commitmen
             await ValidateDateOfChange(command, validationResult);
 
             await _commitmentsApi.PatchEmployerApprenticeship(command.EmployerAccountId, command.ApprenticeshipId, apprenticeshipSubmission);
+
+            var apprenticeship = await _commitmentsApi.GetEmployerApprenticeship(command.EmployerAccountId, command.ApprenticeshipId);
```
Comment (lazcool) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/UpdateApprenticeshipStatus/UpdateApprenticeshipStatusCommandHandler.cs]: We could check to see if SendEmail is enabled in config before fetching the apprenticeship, but as we always run with SendEmail enabled, it's probably not worth it!

File: src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/UpdateApprenticeshipStatus/UpdateApprenticeshipStatusCommandHandler.cs [hunk 3]
```diff
@@ -52,6 +60,9 @@ public UpdateApprenticeshipStatusCommandHandler(IEmployerCommitmentApi commitmen
...
+            await _providerEmailNotificationService.SendProviderApprenticeshipStopNotification(apprenticeship);
```
Comment (lazcool) [inline src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/UpdateApprenticeshipStatus/UpdateApprenticeshipStatusCommandHandler.cs]: Shouldn't we check that 'command.ChangeType == ChangeStatusType.Stop' before sending a stop notification?

**Comment by satya-sanikommu** [comment]
Review and Testing completed. Ready to merge

**Comment by satya-sanikommu** [comment]
Reverted this PR because this has been Squash merged and another branch has been created from this, which now has conflicts. Will be just merging this again in a different PR

**Pull request merged**